added custom icons to view attains map

hillarymarler opened the pull request:
Add custom icons to TADA_ViewATTAINS when the data contain information about the source of the ML/AU crosswalk. The plain circles are from TADA_CreateATTAINSAUMLCrosswalk, the circles with a check mark are from ATTAINS, and the circles with user icon are from a user supplied ref.

What should I do about a legend? Should I try to add the icons to the current legend? Add an additional legend box?

Also, should the different icons for the source refs be a param that can be turned off?

<img width="357" height="275" alt="image" src="https://github.com/user-attachments/assets/e5385ada-baa4-4881-83c8-5a35f1bc89e7" />

hillarymarler committed:
added custom icons to view attains map

hillarymarler committed:
add option to turn off custom icons

hillarymarler committed:
very rough draft of additional legend for crosswalk source

need to edit font, sizing, and placement of new legend

hillarymarler commented:
<img width="533" height="342" alt="image" src="https://github.com/user-attachments/assets/a19d7b54-ecf5-4de2-89a4-934e5a309ccf" />
What about this as a title for the new legend?

hillarymarler committed:
add leaflegend and create crosswalk legend for viewATTAINS

hillarymarler commented:
<img width="266" height="185" alt="image" src="https://github.com/user-attachments/assets/7f414caf-3189-4067-85c5-a5d387192402" />
 We could also do something like this.

hillarymarler committed:
progress on single legend option

hillarymarler committed:
legend updates, added description of new legend to vignette

cristinamullin committed:
Merge branch 'develop' into 646-legend-for-color-coding-of-wqp-observations-and-what-they-correspond-to-in-tada_viewattains

cristinamullin committed:
Update TADA_GetATTAINSByAUID.Rd

cristinamullin committed:
Update TADAModule2.Rmd

cristinamullin commented:
Do you think there should be one more option in the legend to handle cases where add_catch = TRUE (see screenshot for inspiration). Also, I ran into this error when trying to re-run this line after changing add_catch from FALSE to TRUE in the ExampleMod2Workflow.Rmd.

```r
> MT.AUMLRef <- TADA_CreateAUMLCrosswalk(tada.MT.clean,
+                                        au_ref = user.supplied.cw,
+                                        org_id = "MTDEQ",
+                                        add_catch = TRUE,
+                                        batch_upload = TRUE
+ )
[1] "TADA_CreateAUMLCrosswalk: fetching geospatial data for user-supplied crosswalk."
Error in `dplyr::rename()`:
! Can't rename columns that don't exist.
✖ Column `waterType` doesn't exist.
Run `rlang::last_trace()` to see where the error occurred.
```

<img width="1840" height="813" alt="image" src="https://github.com/user-attachments/assets/3de930ec-2ae4-4adc-af5c-426f6b679e71" />

hillarymarler commented:
You are right, there should be another entry in the legend to handle add_catch  = TRUE. I will work on fixing the error you found too.

cristinamullin committed:
dont eval

cristinamullin committed:
Update ExampleMod2Workflow.Rmd

cristinamullin committed:
Update ExampleMod2Workflow.Rmd

cristinamullin commented:
> You are right, there should be another entry in the legend to handle add_catch = TRUE. I will work on fixing the error you found too.

We had also talked about pulling that part out of the function & creating a separate function for the add_catch = TRUE functionality. This could be a separate issue to work on later.

cristinamullin committed:
Update ExampleMod2Workflow.Rmd

cristinamullin committed:
Update WQX3-Migration.Rmd

hillarymarler commented:
> Should the icons always be an option even if all are from the same source? For example, see the example at the bottom of TADAModule2.Rmd. In this case, even though ref_icons = TRUE, the source must not be provided in .data because it does not include the icon on the map. Should the source always be available?

That's a good question. The source ref could be added in each function not just the overall TADA_CreateAUMLCrosswalk function.

hillarymarler committed:
Update ExampleMod2Workflow.Rmd

cristinamullin committed:
Update TADAModule1_AdvancedTraining.Rmd

hillarymarler committed:
Merge branch '646-legend-for-color-coding-of-wqp-observations-and-what-they-correspond-to-in-tada_viewattains' of https://github.com/USEPA/EPATADA into 646-legend-for-color-coding-of-wqp-observations-and-what-they-correspond-to-in-tada_viewattains

hillarymarler committed:
added notes

wokenny13 committed:
Edits to TADA_CreateAUMLCrosswalk to handle NA ATTAINS.WaterType

Added a few lines of code to handle the blank NAs with a user supplied crosswalk or from attains crosswalk.

minor text edits.

was trying to test handling logic if add_catch = TRUE. Once this is fixed I can check to make sure if the edits in this PR will run as intended.

wokenny13 committed:
test for mismatching ATTAINS.WaterType

hillarymarler committed:
Merge branch '646-legend-for-color-coding-of-wqp-observations-and-what-they-correspond-to-in-tada_viewattains' of https://github.com/USEPA/EPATADA into 646-legend-for-color-coding-of-wqp-observations-and-what-they-correspond-to-in-tada_viewattains

hillarymarler committed:
added water types to ExampleMod2 user ref

hillarymarler commented on vignettes/ExampleMod2Workflow.Rmd:
````diff
@@ -243,11 +217,11 @@ to AU are prioritized as follows:
+1)  TADA has developed a function called `TADA_CreateWaterUseRef` which imports a crosswalk of existing uses by water type from ExpertQuery for a specified organization. It can be leveraged within `TADA_CreateUseAURef` to assign any new AU the applicable uses for that water type. Users may choose to edit the `TADA_CreateWaterUseRef` crosswalk if needed.
 
 ```{r assign.use.to.new.AU.with.waterUseRef}
 MT.UseAURef.WaterUseRef <- TADA_CreateUseAURef(
````
I haven't figured out why yet but for some reason this is adding a mostly empty row at the end. Is this what should happen? 
<img width="563" height="70" alt="image" src="https://github.com/user-attachments/assets/9204055b-2691-4251-8a28-07bbc135d589" />

wokenny13 replied:
This empty row occurs as there are sites that do not get assigned to any AU. I can remove the empty row though as I don't know how much value adding this would have.

hillarymarler commented on vignettes/ExampleMod2Workflow.Rmd:
````diff
@@ -257,7 +231,7 @@ MT.UseAURef.WaterUseRef <- TADA_CreateUseAURef(
+2)  Users may manually assign use names to any new AUs if desired rather than using `TADA_CreateWaterUseRef`. For example:
 
 ```{r assign.use.to.new.AU.manual}
 MT.UseAURef.manual <- MT.UseAURef %>%
````
The extra, mostly empty row is retained here as well.

hillarymarler committed:
removed unnecessary col rename from example user ref

hillarymarler commented on R/GeospatialFunctions.R:
```diff
@@ -3435,6 +3534,41 @@ TADA_CreateAUMLCrosswalk <- function(.data, au_ref = NULL,
       user.matches <- spsUtil::quiet(
         TADA_GetATTAINSByAUID(au.ref.mls, au_ref = au_ref, add_catch = add_catch)
       )
+      
```
Do we need to add an if statement here? So that is will run if there are new AUs (can't be retrieved from ATTAINS) when add_catch = FALSE? Or should it always run when add_catch = FALSE?

hillarymarler commented:
> > <img alt="image" width="227" height="80" src="https://private-user-images.githubusercontent.com/152432687/486819397-f6a067ae-3f86-46bf-bb1f-e5a25700f068.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTczNDIyODMsIm5iZiI6MTc1NzM0MTk4MywicGF0aCI6Ii8xNTI0MzI2ODcvNDg2ODE5Mzk3LWY2YTA2N2FlLTNmODYtNDZiZi1iYjFmLWU1YTI1NzAwZjA2OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwOFQxNDMzMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOTVkNDI3ZDVjNDM1ZmY5NGY0ZTBiYTQ1MDU0Mzc4YjU0M2U5ZTUxMjY4OTYzOGQyZTAwNGRkZmRkN2RkOWI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.vgFA8dShCOuolKe88983IFcmt4_9riWfgb-c3TZISig"> For the user supplied ref in TADA_CreateAUMLCrosswalk should the ATTAINS prefix be required? Right now water type is ATTAINS.WaterType, but the other cols do not have the prefix. It seems like it should be consistent. I am leaning towards no prefix for the user supplied ref as it isn't coming from ATTAINS. What do you think @wokenny13 and @cristinamullin ?
> 
> If the function accepts a user supplied input file with or without the prefix, would the function output change the names to ATTAINS.WaterType, ATTAINS. AssessmentUnitIdentifier, and TADA.MonitoringLocationIdentifier?

Yes, the function output would change them so that all the AU/ML/Water Type data from all sources (ATTAINS, User-supplied, TADA_CreateATTAINSAUMLCrosswalk) would be displayed in the same cols in the output of TADA_CreateAUMLCrosswalk

cristinamullin commented:
> > > <img alt="image" width="227" height="80" src="https://private-user-images.githubusercontent.com/152432687/486819397-f6a067ae-3f86-46bf-bb1f-e5a25700f068.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTczNDIyODMsIm5iZiI6MTc1NzM0MTk4MywicGF0aCI6Ii8xNTI0MzI2ODcvNDg2ODE5Mzk3LWY2YTA2N2FlLTNmODYtNDZiZi1iYjFmLWU1YTI1NzAwZjA2OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwOFQxNDMzMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOTVkNDI3ZDVjNDM1ZmY5NGY0ZTBiYTQ1MDU0Mzc4YjU0M2U5ZTUxMjY4OTYzOGQyZTAwNGRkZmRkN2RkOWI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.vgFA8dShCOuolKe88983IFcmt4_9riWfgb-c3TZISig"> For the user supplied ref in TADA_CreateAUMLCrosswalk should the ATTAINS prefix be required? Right now water type is ATTAINS.WaterType, but the other cols do not have the prefix. It seems like it should be consistent. I am leaning towards no prefix for the user supplied ref as it isn't coming from ATTAINS. What do you think @wokenny13 and @cristinamullin ?
> > 
> > 
> > If the function accepts a user supplied input file with or without the prefix, would the function output change the names to ATTAINS.WaterType, ATTAINS. AssessmentUnitIdentifier, and TADA.MonitoringLocationIdentifier?
> 
> Yes, the function output would change them so that all the AU/ML/Water Type data from all sources (ATTAINS, User-supplied, TADA_CreateATTAINSAUMLCrosswalk) would be displayed in the same cols in the output of TADA_CreateAUMLCrosswalk

Okay, the ATTAINS prefix could be optional then (included or excluded) for the user supplied ref. Would that logic apply to all these columns? For example, would it accept any of the following... ATTAINS.WaterType, ATTAINS. AssessmentUnitIdentifier, TADA.MonitoringLocationIdentifier, WaterType, AssessmentUnitIdentifier, and MonitoringLocationIdentifier?

hillarymarler commented:
> > > > <img alt="image" width="227" height="80" src="https://private-user-images.githubusercontent.com/152432687/486819397-f6a067ae-3f86-46bf-bb1f-e5a25700f068.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTczNDIyODMsIm5iZiI6MTc1NzM0MTk4MywicGF0aCI6Ii8xNTI0MzI2ODcvNDg2ODE5Mzk3LWY2YTA2N2FlLTNmODYtNDZiZi1iYjFmLWU1YTI1NzAwZjA2OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwOFQxNDMzMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOTVkNDI3ZDVjNDM1ZmY5NGY0ZTBiYTQ1MDU0Mzc4YjU0M2U5ZTUxMjY4OTYzOGQyZTAwNGRkZmRkN2RkOWI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.vgFA8dShCOuolKe88983IFcmt4_9riWfgb-c3TZISig"> For the user supplied ref in TADA_CreateAUMLCrosswalk should the ATTAINS prefix be required? Right now water type is ATTAINS.WaterType, but the other cols do not have the prefix. It seems like it should be consistent. I am leaning towards no prefix for the user supplied ref as it isn't coming from ATTAINS. What do you think @wokenny13 and @cristinamullin ?
> > > 
> > > 
> > > If the function accepts a user supplied input file with or without the prefix, would the function output change the names to ATTAINS.WaterType, ATTAINS. AssessmentUnitIdentifier, and TADA.MonitoringLocationIdentifier?
> > 
> > 
> > Yes, the function output would change them so that all the AU/ML/Water Type data from all sources (ATTAINS, User-supplied, TADA_CreateATTAINSAUMLCrosswalk) would be displayed in the same cols in the output of TADA_CreateAUMLCrosswalk
> 
> Okay, the ATTAINS prefix could be optional then (included or excluded) for the user supplied ref. Would that logic apply to all these columns? For example, would it accept any of the following... ATTAINS.WaterType, ATTAINS. AssessmentUnitIdentifier, TADA.MonitoringLocationIdentifier, WaterType, AssessmentUnitIdentifier, and MonitoringLocationIdentifier?

Optional sounds good to me! I can make that change.

hillarymarler committed:
added internal function in utilities to check cols which may or may not have prefixes in user ref

still need to make some updates in TADA_GetATTAINSByAUID

hillarymarler committed:
documentation updates

hillarymarler committed:
changes to make col names agree

still working on add_catch = TRUE issues

hillarymarler committed:
added notes

hillarymarler committed:
TADA_ViewATTAINS fitBounds bug fix

hillarymarler committed:
removed duplicate set.popup code

hillarymarler committed:
styler updates

hillarymarler committed:
Update ExampleMod2Workflow.Rmd

removed : from description (causing scanner error)

hillarymarler committed:
Update ExampleMod2Workflow.Rmd

updated ref branch

hillarymarler committed:
removed water type from ATTAINS batch upload output

hillarymarler committed:
Update Utilities.R

added missing global vars

hillarymarler committed:
Update Figures.R

add na.rm = TRUE to max/min for lat/long in TADA_OverviewMap

hillarymarler committed:
icon updates

hillarymarler committed:
bug fix for TADA_CreateATTAINSAUMLCrosswalk, add additional catcment icon

hillarymarler committed:
added ATTAINS col renaming function to utilities

hillarymarler committed:
Update GeospatialFunctions.R

add rename function

cristinamullin committed:
update docs

Updated roxygen2 and added RoxygenNote: 7.3.3 to description. Ran devtools::document()

cristinamullin committed:
update refs

cristinamullin committed:
update example data

hillarymarler committed:
Update ExampleMod2Workflow.Rmd

hillarymarler committed:
Merge branch '646-legend-for-color-coding-of-wqp-observations-and-what-they-correspond-to-in-tada_viewattains' of https://github.com/USEPA/EPATADA into 646-legend-for-color-coding-of-wqp-observations-and-what-they-correspond-to-in-tada_viewattains

hillarymarler committed:
add gray filled catchments to legend

cristinamullin committed:
update checkColNames

typos

hillarymarler committed:
viewATTAINS and createAUMLcrosswalk updates

hillarymarler committed:
Update GeospatialFunctions.R

hillarymarler committed:
updated view map to display nhd catch in legend

hillarymarler committed:
bug fixes

hillarymarler committed:
documentation updates

hillarymarler commented:
All checks are now passing. The changes we discussed to the TADA_ViewATTAINS map are incorporated and ready for review.

hillarymarler commented:
> I am looking to run TADA_CreateAUMLCrosswalk() if an org has no user supplied crosswalk. I am unsure if I am doing something wrong, but I run into an error when I run the code below:
> 
> ```
> ## retrieve data for your org
> tada.data <- TADA_DataRetrieval(
>   startDate = "2010-01-01",
>   endDate = "2025-12-31",
>   organization = "PENOBSCOTINDIANNATIONDNR",
>   ask = FALSE
> )
> 
> # clean up data set (minimal)
> tada.data.clean <- tada.data %>%
>   TADA_RunKeyFlagFunctions() %>%
>   TADA_SimpleCensoredMethods() %>%
>   TADA_HarmonizeSynonyms()
> 
> tada.AUMLRef <- TADA_CreateAUMLCrosswalk(tada.data.clean,
>   #au_ref = user.supplied.cw,
>   org_id = "MEDEP",
>   add_catch = FALSE,
>   nhd_catch = TRUE,
>   batch_upload = TRUE)
> ```
> 
Good catch! I was so focused on getting it set up to work for user, ATTAINS, and TADA_CreateATTAINSAUML all together that I did not correctly account for when au_ref = NULL. I will make this bug fix on the merged branch.

cristinamullin commented:
> > I am looking to run TADA_CreateAUMLCrosswalk() if an org has no user supplied crosswalk. I am unsure if I am doing something wrong, but I run into an error when I run the code below:
> > ```
> > ## retrieve data for your org
> > tada.data <- TADA_DataRetrieval(
> >   startDate = "2010-01-01",
> >   endDate = "2025-12-31",
> >   organization = "PENOBSCOTINDIANNATIONDNR",
> >   ask = FALSE
> > )
> > 
> > # clean up data set (minimal)
> > tada.data.clean <- tada.data %>%
> >   TADA_RunKeyFlagFunctions() %>%
> >   TADA_SimpleCensoredMethods() %>%
> >   TADA_HarmonizeSynonyms()
> > 
> > tada.AUMLRef <- TADA_CreateAUMLCrosswalk(tada.data.clean,
> >   #au_ref = user.supplied.cw,
> >   org_id = "MEDEP",
> >   add_catch = FALSE,
> >   nhd_catch = TRUE,
> >   batch_upload = TRUE)
> > ```
> 
> Good catch! I was so focused on getting it set up to work for user, ATTAINS, and TADA_CreateATTAINSAUML all together that I did not correctly account for when au_ref = NULL. I will make this bug fix on the merged branch.

can org_id be NULL as well?

hillarymarler commented:
> can org_id be NULL as well?

Yes, I think it should be able to. I am adding that as part of the updates along with the au_ref ones. Currently, if org_id = NULL, ATTAINS will not be checked for a ML/AU crosswalk at all. 

The other option would be that a NULL org_id would lead to checking all organizations'  AU/ML crosswalks for ML matches. This could be done using the ExpertQuery national Assessment Units with Monitoring Locations profile via rExpertQuery functions. It would take longer to load the national profile, but because it is just looking for exact character string matches (no geospatial joins), I don't think the increase in run time would be too crazy. Or this could be an additional option (org_id = "all"?)

What do you think?

cristinamullin commented on R/GeospatialFunctions.R:
```diff
@@ -3357,13 +3434,26 @@ TADA_RandomTestingData <- function(number_of_days = 1, choose_random_state = FAL
-#' and MonitoringLocationIdentifier.
+#' @param au_ref Optional. A user-supplied df with the columns AssessmentUnitIdentifier,
+#' MonitoringLocationIdentifier and WaterType.
 #' @param org_id Organization id to match AUs.
```
I tried leaving this blank but got an error. Is this required or if left blank, can it match with any AUs from ATTAINS regardless of org? This may be useful for some tribes near the border of two states that are interested in using the state AUs for their own analyses. Or for regional EPA folks or other researchers who are looking at data from multiple states.

Can users specify multiple?

cristinamullin replied:
Suggest to include where to find the ATTIANS org ids here as well (example R code or link) 

Working on this now: 

#' Character argument. The ATTAINS organization identifier. AUs will
#' only be matched to the specified organization(s). A list of organization 
#' identifiers can be found 
#' by downloading the ATTAINS Domains Excel file: 
#' https://www.epa.gov/system/files/other-files/2025-02/domains_2025-02-25.xlsx.
#' Organization identifiers are listed in the "OrgName" tab. The "code" column 
#' contains the organization identifiers that should be used for this param.

cristinamullin committed:
Update test-CensoredDataSuite.R

cristinamullin committed:
update documentation

cristinamullin commented on R/GeospatialFunctions.R:
```diff
@@ -2109,15 +2058,21 @@ TADA_GetATTAINSByAUID <- function(.data, au_ref = NULL, add_catch = FALSE) {
-#' attains_data <- TADA_CreateATTAINSAUMLCrosswalk(tada_data,
+#' 
+#' # Match AUs using all available methods
+#' all_sources <- TADA_CreateAUMLCrosswalk(tada_data, org_id = "21NEV1")
```
I tried to add this example but ran into a bug. I think it is not able to handle when the user supplied ref is blank.

```r
Error in `dplyr::filter()`:
ℹ In argument: `!TADA.MonitoringLocationIdentifier %in%
  au.ref.mls$TADA.MonitoringLocationIdentifier`.
Caused by error:
! object 'au.ref.mls' not found
---
Backtrace:
     ▆
  1. ├─EPATADA::TADA_CreateAUMLCrosswalk(tada_data, org_id = "21NEV1")
  2. │ └─... %>% ... at EPATADA/R/GeospatialFunctions.R:3643:5
  3. ├─dplyr::mutate(., TADA.AURefSource = "ATTAINS Crosswalk")
  4. ├─dplyr::filter(...)
  5. ├─dplyr:::filter.data.frame(...)
  6. │ └─dplyr:::filter_rows(.data, dots, by)
  7. │   └─dplyr:::filter_eval(...)
  8. │     ├─base::withCallingHandlers(...)
  9. │     └─mask$eval_all_filter(dots, env_filter)
 10. │       └─dplyr (local) eval()
 11. ├─TADA.MonitoringLocationIdentifier %in% ...
 12. └─base::.handleSimpleError(...)
 13.   └─dplyr (local) h(simpleError(msg, call))
 14.     └─rlang::abort(message, class = error_class, parent = parent, call = error_call)
 15. ```

hillarymarler replied:
I fixed this in the mod 3 branch. Now it will run if no au_ref is provided.

cristinamullin committed:
update tests
